packages/hash/api/src/model/entityType.model.ts

```diff
@@ -103,17 +142,30 @@ class __EntityType {
     return new EntityType(entityType);
   }
 
-  static async updateSchema(
+  async update(
     client: DBClient,
     params: {
       accountId: string;
       createdByAccountId: string;
-      entityId: string;
-      schema: JSONObject;
+      schema: Record<string, any>;
       updatedByAccountId: string;
     },
   ) {
-    const updatedDbEntityType = await client.updateEntityType(params);
+    const {
+      schema: { title, description },
+    } = params;
+
+    const schema = await EntityType.validateJsonSchema(client, {
+      name: title,
+      schema: params.schema,
+      description,
+    });
+
+    const updatedDbEntityType = await client.updateEntityType({
+      ...params,
+      entityId: this.entityId,
+      schema,
+    });
```

Comment:
**Blocking:** Because this is now no longer a static function, we need to update the member variables that are affected by the db update. We could either do this individually by setting things like `properties`, `updatedAt`, etc - or we could make use of lodash `merge`:

```suggestion
    const updatedDbEntityType = await client.updateEntityType({
      ...params,
      entityId: this.entityId,
      schema,
    });
    
    merge(this, new EntityType(updatedDbEntityType));
    
    return this;
```

This is what we are also currently doing in `Entity.updateProperties`.

Reply:
Of course, sorry for missing that! Thank you :)

Reply:
I think my instinct was to not mutate the calling object, but since we do this throughout the model classes, it will be changed here :)

Reply:
What would be the reasoning for not mutating the calling object? I feel if we don't mutate it we risk that the caller of the model class instance makes use of outdated data (for instance the incorrect `updatedAt`, the outdated `schema`, etc).

I appreciate that this wouldn't be an issue if we were to consistently use the return value of an update function, but this will be hard to enforce and could therefore quickly lead to bugs.

Reply:
Yes, it makes sense for our model classes we're doing it this way. My instinct stems from other projects where it lead to some unexpected behaviour, so it was more general instinct - not bound to our context.

I don't think those worries exist for our model classes since they are short lived and owned by a single resolver anyway :)